src/APIBasedAuth.ts

```diff
@@ -1,11 +1,64 @@
 import axios, { AxiosInstance, AxiosResponse } from 'axios';
-import { window } from './globals';
 import { DEFAULT_BASE_URL, formatAxiosError } from './utils/helper';
 
 export const API_AUTH_STORAGE_KEY = 'lo-app-tools-api-auth';
 const SESSION_KEYS = ['session', 'username'];
 const TOKEN_KEYS = ['accessToken', 'idToken', 'refreshToken'];
 
+/**
+ * This class performs basic API based authentication based on our apps-auth repo
+ *
+ * Example storage differences
+ *
+ * Web:
+ * const auth = new APIBasedAuth({
+ *   clientId: 'client_id',
+ *   storage: {
+ *     getItem: (key) => Promise.resolve(window.localStorage.getItem(key)),
+ *     removeItem: (key) => Promise.resolve(window.localStorage.removeItem(key)),
+ *     setItem: (key, value) => Promise.resolve(window.localStorage.setItem(key, value)),
+ *   },
+ *   storageKeys: {
+ *     accessToken: 'custom_access_token_key',
+ *     idToken: 'custom_identity_token_key',
+ *     refreshToken: 'custom_refresh_token_key',
+ *     session: 'custom_session_key',
+ *     username: 'custom_username_key',
+ *   },
+ * });
+ *
+ * Mobile (Expo):
+ * import * as SecureStore from 'expo-secure-store';
+ *
+ * const auth = new APIBasedAuth({
+ *   clientId: 'client_id',
+ *   storage: {
+ *     getItem: async (key) => {
+ *       const value = await SecureStore.getItemAsync(key);
+ *       return value;
+ *     },
+ *     removeItem: async (key) => {
+ *       await SecureStore.deleteItemAsync(key);
+ *     },
+ *     setItem: async (key, value) => {
+ *       await SecureStore.setItemAsync(key, value);
+ *     },
+ *   },
+ *   storageKeys: {
```

Comment:
The `storageKeys` seems like would be reasonble to provide defaults.

Reply:
the default keys are inline so the user has the ability to currently use our default naming for any keys and use their own custom naming for others at their discretion. there are the defaults currently
```
{
  accessToken: 'lo-app-tools-api-auth.accessToken',
  idToken: 'lo-app-tools-api-auth.idToken',
  refreshToken: 'lo-app-tools-api-auth.refreshToken',
  session: 'lo-app-tools-api-auth.session',
  username: 'lo-app-tools-api-auth.username',
}
```
if the users wanted to just have custom values for session/username they could without modifying the default for the other 3, but if we defaulted storageKeys then they would have to provide all 5 keys in the class initialization

Reply:
I think you're saying: defaults will be used for any unspecified keys. If that's right, I think we're in good shape!